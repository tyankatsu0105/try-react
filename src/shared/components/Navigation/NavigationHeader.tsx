import React, { FunctionComponent } from 'react';
import clsx from 'clsx';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { AppBar, Toolbar, IconButton, Typography } from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';

const drawerWidth = 8 * 30;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    navigationHeader: {
      zIndex: theme.zIndex.drawer + 1,
      transition: theme.transitions.create(['width', 'margin'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
    },
    'navigationHeader-shift': {
      marginLeft: drawerWidth,
      width: `calc(100% - ${drawerWidth}px)`,
      transition: theme.transitions.create(['width', 'margin'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.enteringScreen,
      }),
    },
    'navigationHeader-MenuButton': {
      marginRight: 36,
    },
    'navigationHeader-MenuButton-hide': {
      display: 'none',
    },
  })
);

interface Props {
  open: boolean;
  handleDrawerOpen: () => void;
}

export const NavigationHeader: FunctionComponent<Props> = (props) => {
  const classes = useStyles();

  const { open, handleDrawerOpen } = props;

  return (
    <AppBar
      className={clsx(classes['navigationHeader'], {
        [classes['navigationHeader-shift']]: open,
      })}
    >
      <Toolbar>
        <IconButton
          color="inherit"
          aria-label="open drawer"
          onClick={handleDrawerOpen}
          edge="start"
          className={clsx(classes['navigationHeader-MenuButton'], {
            [classes['navigationHeader-MenuButton-hide']]: open,
          })}
        >
          <MenuIcon />
        </IconButton>
        <Typography variant="h6" noWrap>
          Mini variant drawer
        </Typography>
      </Toolbar>
    </AppBar>
  );
};
