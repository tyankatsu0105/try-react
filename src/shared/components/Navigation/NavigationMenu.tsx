import React, { FunctionComponent } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

import {
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  IconButton,
  Link,
} from '@material-ui/core';
import InboxIcon from '@material-ui/icons/MoveToInbox';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';

const menus = [
  {
    path: '/',
    text: 'ホーム',
  },
  {
    path: '/about',
    text: '自己紹介',
  },
  {
    path: '/material',
    text: 'マテリアル',
  },
  {
    path: '/formik-form',
    text: 'formikフォーム',
  },
  {
    path: '/github-query',
    text: 'GitHub Query',
  },
];

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    'navigationMenu-Header': {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-end',
      padding: `0 ${theme.spacing(1)}`,
      ...theme.mixins.toolbar,
    },
  })
);

interface Props {
  handleDrawerClose: () => void;
}

export const NavigationMenu: FunctionComponent<Props> = (props) => {
  const classes = useStyles();
  const { handleDrawerClose } = props;

  return (
    <nav className="navigationMenu">
      <div className={classes['navigationMenu-Header']}>
        <IconButton onClick={handleDrawerClose}>
          <ChevronLeftIcon />
        </IconButton>
      </div>
      <List className="navigationMenu-List">
        {menus.map((menu) => (
          <Link
            key={menu.path}
            component={RouterLink}
            to={menu.path}
            className="navigationMenu-ListItem"
          >
            <ListItem button>
              <ListItemIcon>
                <InboxIcon />
              </ListItemIcon>
              <ListItemText primary={menu.text} />
            </ListItem>
          </Link>
        ))}
      </List>
    </nav>
  );
};
