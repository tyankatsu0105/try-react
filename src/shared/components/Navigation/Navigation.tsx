import React, { FunctionComponent } from 'react';
import clsx from 'clsx';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

import { Drawer, Divider } from '@material-ui/core';

import { NavigationMenu } from './NavigationMenu';
import { NavigationHeader } from './NavigationHeader';

const drawerWidth = 8 * 30;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    navigation: {
      display: 'flex',
    },
    'navigation-Drawer': {
      width: drawerWidth,
      flexShrink: 0,
      whiteSpace: 'nowrap',
    },
    'navigation-Drawer-open': {
      width: drawerWidth,
      transition: theme.transitions.create('width', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.enteringScreen,
      }),
    },
    'navigation-Drawer-close': {
      transition: theme.transitions.create('width', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
      overflowX: 'hidden',
      width: theme.spacing(7) + 1,
    },
    content: {
      flexGrow: 1,
      padding: theme.spacing(3),
      paddingTop: theme.spacing(10),
    },
  })
);

export const Navigation: FunctionComponent = (props) => {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);

  const { children } = props;

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  return (
    <div className={classes.navigation}>
      <NavigationHeader open={open} handleDrawerOpen={handleDrawerOpen} />
      <Drawer
        variant="permanent"
        className={clsx(classes['navigation-Drawer'], {
          [classes['navigation-Drawer-open']]: open,
          [classes['navigation-Drawer-close']]: !open,
        })}
        classes={{
          paper: clsx({
            [classes['navigation-Drawer-open']]: open,
            [classes['navigation-Drawer-close']]: !open,
          }),
        }}
        {...open}
      >
        <NavigationMenu handleDrawerClose={handleDrawerClose} />
        <Divider />
      </Drawer>
      <main className={classes.content}>{children}</main>
    </div>
  );
};
