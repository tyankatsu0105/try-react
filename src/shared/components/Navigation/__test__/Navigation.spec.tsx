import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

import { Navigation } from '../Navigation';

describe('Navigation', () => {
  it('snapshot', () => {
    const { baseElement } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
