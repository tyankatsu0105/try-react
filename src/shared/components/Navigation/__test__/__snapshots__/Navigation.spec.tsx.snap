// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-navigation-1"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-navigationHeader-6 MuiAppBar-colorPrimary mui-fixed"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-navigationHeader-MenuButton-8 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
          >
            Mini variant drawer
          </h6>
        </div>
      </header>
      <div
        class="MuiDrawer-root MuiDrawer-docked makeStyles-navigation-Drawer-2 makeStyles-navigation-Drawer-close-4"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation0 MuiDrawer-paper makeStyles-navigation-Drawer-close-4 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
        >
          <nav
            class="navigationMenu"
          >
            <div
              class="makeStyles-navigationMenu-Header-113"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <ul
              class="MuiList-root navigationMenu-List MuiList-padding"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/"
              >
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root navigationMenu-ListItem MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                      />
                      <path
                        d="M0 0h24v24H0V0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                    >
                      ホーム
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/about"
              >
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root navigationMenu-ListItem MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                      />
                      <path
                        d="M0 0h24v24H0V0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                    >
                      自己紹介
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/material"
              >
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root navigationMenu-ListItem MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                      />
                      <path
                        d="M0 0h24v24H0V0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                    >
                      マテリアル
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </a>
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/formik-form"
              >
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root navigationMenu-ListItem MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H4.99c-1.11 0-1.98.9-1.98 2L3 19c0 1.1.88 2 1.99 2H19c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 12h-4c0 1.66-1.35 3-3 3s-3-1.34-3-3H4.99V5H19v10zm-3-5h-2V7h-4v3H8l4 4 4-4z"
                      />
                      <path
                        d="M0 0h24v24H0V0z"
                        fill="none"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                    >
                      formikフォーム
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </a>
            </ul>
          </nav>
          <hr
            class="MuiDivider-root"
          />
        </div>
      </div>
      <main
        class="makeStyles-content-5"
      />
    </div>
  </div>
</body>
`;
